src/Microsoft.ML/MemoryCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML
+{
+    public class MemoryCollection
+    {
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IList<T> data) where T:class
+        {
+            return new MemoryCollectionLoader<T>(data);
+        }
+
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IEnumerable<T> data) where T : class
+        {
+            return new MemoryCollectionLoader<T>(data);
+        }
+    }
+
+    /// <summary>
+    /// Allows you to convert your memory collection into IDataview.
+    /// </summary>
+    /// <typeparam name="TInput"></typeparam>
+    public class MemoryCollectionLoader<TInput> : ILearningPipelineLoader
```

Comment:
>MemoryCollectionLoader [](start = 17, length = 22)

This class very closely mimics the ListDataView and StreamingDataView.  the key difference between the two being that one is shuffleble and nother is not.  Shoould we also create two top level classes here as well? 
Then we can have proper names like ListLoader and StreamingEnumerationLoader

Reply:
The key difference is implementation of interface. This is new API, I have to create wrappers for it.

---
In reply to: [187227251](https://github.com/dotnet/machinelearning/pull/106#discussion_r187227251) [](ancestors = 187227251)